In [4]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
df = pd.read_csv("FoodBalanceSheets_E_Africa_NOFLAG.csv", encoding = "latin-1")


df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 60943 entries, 0 to 60942
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Area Code     60943 non-null  int64  
 1   Area          60943 non-null  object 
 2   Item Code     60943 non-null  int64  
 3   Item          60943 non-null  object 
 4   Element Code  60943 non-null  int64  
 5   Element       60943 non-null  object 
 6   Unit          60943 non-null  object 
 7   Y2014         59354 non-null  float64
 8   Y2015         59395 non-null  float64
 9   Y2016         59408 non-null  float64
 10  Y2017         59437 non-null  float64
 11  Y2018         59507 non-null  float64
dtypes: float64(5), int64(3), object(4)
memory usage: 5.6+ MB


There are 60943 rows and 12 columns in the data and occupies 5.6mb. There are 3 data types present which is float,int,object.bb

In [ ]:
#display first rows

In [6]:
df.head()

Area Code     Area  Item Code         Item  Element Code  \
0          4  Algeria       2501   Population           511   
1          4  Algeria       2501   Population          5301   
2          4  Algeria       2901  Grand Total           664   
3          4  Algeria       2901  Grand Total           674   
4          4  Algeria       2901  Grand Total           684   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28

In [ ]:
#display last rows

In [7]:
df.tail()

Area Code      Area  Item Code           Item  Element Code  \
60938        181  Zimbabwe       2899  Miscellaneous          5142   
60939        181  Zimbabwe       2899  Miscellaneous           645   
60940        181  Zimbabwe       2899  Miscellaneous           664   
60941        181  Zimbabwe       2899  Miscellaneous           674   
60942        181  Zimbabwe       2899  Miscellaneous           684   

                                      Element             Unit  Y2014  Y2015  \
60938                                    Food      1000 tonnes  42.00  46.00   
60939     Food supply quantity (kg/capita/yr)               kg   3.06   3.33   
60940           Food supply (kcal/capita/day)  kcal/capita/day   3.00   4.00   
60941  Protein supply quantity (g/capita/day)     g/capita/day   0.10   0.11   
60942      Fat supply quantity (g/capita/day)     g/capita/day   0.04   0.05   

       Y2016  Y2017  Y2018  
60938  33.00  19.00  16.00  
60939   2.35   1.33   1.08  
60940   3.00   1.00   1.00  
60941   0.08   0.04   0.04  
60942   0.03   0.02   0.01

In [ ]:
#display 10 random rows of data

In [8]:
df.sample(10)

Area Code                   Area  Item Code                     Item  \
3449          53                  Benin       2613  Grapefruit and products   
1044           4                Algeria       2945                   Offals   
46896        193  Sao Tome and Principe       2949                     Eggs   
34512        133                   Mali       2642                   Cloves   
37504        143                Morocco       2511       Wheat and products   
60534        181               Zimbabwe       2924      Alcoholic Beverages   
48784        197           Sierra Leone       2908              Sugar Crops   
11111         39                   Chad       2536               Sugar cane   
6101         233           Burkina Faso       2582           Maize Germ Oil   
15786         72               Djibouti       2630      Coffee and products   

       Element Code                                 Element             Unit  \
3449           5142                                    Food      1000 tonnes   
1044            674  Protein supply quantity (g/capita/day)     g/capita/day   
46896          5171                     Tourist consumption      1000 tonnes   
34512           645     Food supply quantity (kg/capita/yr)               kg   
37504          5142                                    Food      1000 tonnes   
60534          5611                         Import Quantity      1000 tonnes   
48784          5521                                    Feed      1000 tonnes   
11111          5131                              Processing      1000 tonnes   
6101            664           Food supply (kcal/capita/day)  kcal/capita/day   
15786          5170                               Residuals      1000 tonnes   

         Y2014    Y2015    Y2016    Y2017    Y2018  
3449      0.00     0.00     0.00     0.00     0.00  
1044      0.73     0.71     0.73     0.73     0.68  
46896     0.00     0.00     0.00     0.00     0.00  
34512     0.00     0.00     0.00     0.00     0.00  
37504  6194.00  6286.00  6620.00  6455.00  6507.00  
60534    14.00     9.00     8.00     5.00     6.00  
48784     4.00     5.00     4.00     4.00     0.00  
11111   357.00   381.00   373.00   353.00   358.00  
6101      1.00     0.00     0.00     0.00     0.00  
15786     0.00     0.00     0.00     0.00     0.00

Get a summary descriptive statics

In [10]:
df.describe()

Area Code     Item Code  Element Code          Y2014          Y2015  \
count  60943.000000  60943.000000  60943.000000   59354.000000   59395.000000   
mean     134.265576   2687.176706   3814.856456     134.196282     135.235966   
std       72.605709    146.055739   2212.007033    1567.663696    1603.403984   
min        4.000000   2501.000000    511.000000   -1796.000000   -3161.000000   
25%       74.000000   2562.000000    684.000000       0.000000       0.000000   
50%      136.000000   2630.000000   5142.000000       0.090000       0.080000   
75%      195.000000   2775.000000   5511.000000       8.340000       8.460000   
max      276.000000   2961.000000   5911.000000  176405.000000  181137.000000   

               Y2016          Y2017          Y2018  
count   59408.000000   59437.000000   59507.000000  
mean      136.555222     140.917765     143.758381  
std      1640.007194    1671.862359    1710.782658  
min     -3225.000000   -1582.000000   -3396.000000  
25%         0.000000       0.000000       0.000000  
50%         0.080000       0.100000       0.070000  
75%         8.430000       9.000000       9.000000  
max    185960.000000  190873.000000  195875.000000

We can see different statistical summary of numeric columns. The mean, standard deviation, minimum and maximum values as well as the Percentile. "For numeric data, the result’s index will include count, mean, std, min, max as well as lower, 50 and upper percentiles. By default the lower percentile is 25 and the upper percentile is 75. The 50 percentile is the same as the median."

In [11]:
# summary for categorical columns
df.describe(include="object")

Area                     Item                   Element         Unit
count   60943                    60943                     60943        60943
unique     49                      119                        18            5
top     Kenya  Milk - Excluding Butter  Domestic supply quantity  1000 tonnes
freq     1560                     1262                      5295        40933

We can see that the statistics returned for categorical columns are different. This included the count, number of unique categories, topmost unique category, and the frequency. "For object data (e.g. strings or timestamps), the result’s index will include count, unique, top, and freq. The top is the most common value. The freq is the most common value’s frequency. Timestamps also include the first and last items."

Check for Missing Values

In [12]:
df.isnull().sum()

Area Code          0
Area               0
Item Code          0
Item               0
Element Code       0
Element            0
Unit               0
Y2014           1589
Y2015           1548
Y2016           1535
Y2017           1506
Y2018           1436
dtype: int64

We can see that the dataset contains missing values . The column Y2014,Y2015,Y2016,Y2017,Y2018 contains 1589,1548,1535,1506,1436 missing values. 
Lets treat the missing values

In [15]:
df[df.Y2014.isnull()]

Area Code      Area  Item Code                 Item  Element Code  \
46             4   Algeria       2805    Rice and products          5911   
61             4   Algeria       2513  Barley and products          5911   
88             4   Algeria       2515     Rye and products          5611   
100            4   Algeria       2516                 Oats          5911   
183            4   Algeria       2533       Sweet potatoes          5911   
...          ...       ...        ...                  ...           ...   
60362        181  Zimbabwe       2616            Plantains           664   
60363        181  Zimbabwe       2616            Plantains           674   
60364        181  Zimbabwe       2616            Plantains           684   
60623        181  Zimbabwe       2732   Mutton & Goat Meat          5911   
60646        181  Zimbabwe       2734         Poultry Meat          5911   

                                      Element             Unit  Y2014  Y2015  \
46                            Export Quantity      1000 tonnes    NaN    0.0   
61                            Export Quantity      1000 tonnes    NaN    0.0   
88                            Import Quantity      1000 tonnes    NaN    NaN   
100                           Export Quantity      1000 tonnes    NaN    NaN   
183                           Export Quantity      1000 tonnes    NaN    NaN   
...                                       ...              ...    ...    ...   
60362           Food supply (kcal/capita/day)  kcal/capita/day    NaN    NaN   
60363  Protein supply quantity (g/capita/day)     g/capita/day    NaN    NaN   
60364      Fat supply quantity (g/capita/day)     g/capita/day    NaN    NaN   
60623                         Export Quantity      1000 tonnes    NaN    NaN   
60646                         Export Quantity      1000 tonnes    NaN    NaN   

       Y2016  Y2017  Y2018  
46       0.0    0.0    0.0  
61       0.0    NaN    0.0  
88       NaN    NaN    0.0  
100      0.0    0.0    0.0  
183      0.0    0.0    0.0  
...      ...    ...    ...  
60362    NaN    0.0    NaN  
60363    NaN    0.0    NaN  
60364    NaN    0.0    NaN  
60623    NaN    0.0    NaN  
60646    0.0    NaN    NaN  

[1589 rows x 12 columns]

Let us replace the missing values of the Y2014 Column with the Mode value of that entire column.

In [18]:
modeValue = df['Y2014'].mode()
modeValue[0]

0.0

In [24]:
df.dropna(subset = ['Y2015'], inplace=True)
df.dropna(subset = ['Y2016'], inplace=True)
df.dropna(subset = ['Y2017'], inplace=True)
df.dropna(subset = ['Y2018'], inplace=True)

In [25]:
df['Y2014'].fillna(value=modeValue[0], inplace = True)
df.isnull().sum()

Area Code       0
Area            0
Item Code       0
Item            0
Element Code    0
Element         0
Unit            0
Y2014           0
Y2015           0
Y2016           0
Y2017           0
Y2018           0
dtype: int64

# Query the Data to answer Analytic Question
Now that we have a clean data, we can proceed to making analysis

In [ ]:
1. #How many unique area and item do we have in our data?
 #Ans :49

In [31]:
df["Area"].value_counts()

Kenya                          1527
Senegal                        1440
Egypt                          1439
Nigeria                        1439
Côte d'Ivoire                  1437
Uganda                         1423
Madagascar                     1415
Cameroon                       1415
Zambia                         1397
Ethiopia                       1393
South Africa                   1392
United Republic of Tanzania    1389
Mozambique                     1370
Malawi                         1368
Rwanda                         1360
Morocco                        1360
Angola                         1358
Botswana                       1349
Burkina Faso                   1346
Congo                          1342
Zimbabwe                       1340
Guinea                         1333
Mauritius                      1324
Niger                          1323
Ghana                          1313
Tunisia                        1295
Algeria                        1255
Namibia                     

In [37]:
df["Item"].value_counts()

Milk - Excluding Butter     1240
Eggs                        1100
Miscellaneous                883
Cereals - Excluding Beer     709
Oilcrops                     695
                            ... 
Grand Total                  135
Animal Products              135
Vegetal Products             135
Sugar non-centrifugal        113
Population                    90
Name: Item, Length: 119, dtype: int64

In [ ]:
#There are quite many of them

In [51]:
2. #What is the sum of plaintain distributed from Y2014 to Y2018?

df[df['Item'] == 'Plantains'].groupby('Item').sum()

Area Code  Item Code  Element Code     Y2014    Y2015     Y2016  \
Item                                                                         
Plantains      44344     868512       1148743  64643.48  64987.0  62985.69   

              Y2017     Y2018  
Item                           
Plantains  63595.19  65170.65

In [73]:

country = df.loc[(df.Area == "Nigeria") & (df["Item"] == "Wine")]
country.sort_values(by= "Item",ascending=False).head(5)

Area Code     Area  Item Code  Item  Element Code  \
44026        159  Nigeria       2655  Wine          5511   
44027        159  Nigeria       2655  Wine          5611   
44028        159  Nigeria       2655  Wine          5072   
44029        159  Nigeria       2655  Wine          5911   
44030        159  Nigeria       2655  Wine          5301   

                        Element         Unit  Y2014  Y2015  Y2016  Y2017  \
44026                Production  1000 tonnes    0.0    0.0    0.0    0.0   
44027           Import Quantity  1000 tonnes   39.0   30.0   17.0   22.0   
44028           Stock Variation  1000 tonnes    7.0   -3.0  -10.0    0.0   
44029           Export Quantity  1000 tonnes    0.0    0.0    0.0    0.0   
44030  Domestic supply quantity  1000 tonnes   32.0   33.0   27.0   22.0   

       Y2018  
44026    0.0  
44027   25.0  
44028    0.0  
44029    0.0  
44030   25.0

In [69]:
3. #What is the total sum of eggs per unit distributed in each year?


df[df['Item'] == 'Eggs'].groupby('Unit').sum()

Area Code  Item Code  Element Code     Y2014     Y2015  \
Unit                                                                      
1000 tonnes         100054    2106410       3944296  19614.00  20442.00   
g/capita/day         24080     512370        122220     99.92    103.44   
kcal/capita/day      12040     256185         59760    654.00    680.00   
kg                   12040     256185         58050    189.44    196.04   

                    Y2016     Y2017     Y2018  
Unit                                           
1000 tonnes      19490.00  20792.00  19746.00  
g/capita/day        97.84    100.42     94.14  
kcal/capita/day    644.00    660.00    620.00  
kg                 184.98    189.32    178.34

In [63]:
4. #Which area had the lowest sum in 2015?
#Ans : Ethiopia and Sudan

df[['Y2015', 'Area']].groupby('Area').sum().sort_values('Y2015', ascending=True)

Y2015
Area                                   
Ethiopia PDR                       0.00
Sudan (former)                     0.00
Comoros                           41.34
Seychelles                       358.20
Sao Tome and Principe          12386.93
Cabo Verde                     14538.93
Lesotho                        17959.84
Guinea-Bissau                  17990.05
Gambia                         19661.39
Botswana                       23024.85
Djibouti                       23686.16
Gabon                          28367.30
Liberia                        28680.83
Central African Republic       28885.34
Namibia                        30686.90
Congo                          40249.51
Togo                           47829.59
Mauritius                      53371.36
Eswatini                       54044.89
Mauritania                     61906.99
Sierra Leone                   62521.13
Chad                           65835.88
Rwanda                         70138.83
Zimbabwe                       74041.79
Senegal                        91939.14
Guinea                         92803.39
Zambia                         96191.00
Burkina Faso                  102698.22
Benin                         108167.13
Niger                         117826.65
Madagascar                    126673.90
Tunisia                       128199.41
Mali                          132445.58
Mozambique                    158218.17
Malawi                        167974.66
Angola                        209565.67
Côte d'Ivoire                 211583.29
Uganda                        220595.25
Cameroon                      225193.72
Sudan                         230508.54
Kenya                         265501.19
Ghana                         311092.13
Algeria                       324048.40
United Republic of Tanzania   341919.46
Morocco                       412473.80
Ethiopia                      429105.04
South Africa                  468327.18
Egypt                         868209.73
Nigeria                      1414166.83

In [74]:
5. #What is the total sum of rice and prooducts distributed in 2014,2015,2016,2017,2018?

df[df['Item'] == 'Rice and products'].groupby('Item').sum()

Area Code  Item Code  Element Code      Y2014     Y2015  \
Item                                                                         
Rice and products      86847    1831665       2648617  155093.23  150656.0   

                       Y2016      Y2017      Y2018  
Item                                                
Rice and products  161149.98  166354.09  167186.55

In [77]:
6. # Which area has the highest value counts?
Ans:Kenya
    
df["Area"].value_counts()

Kenya                          1527
Senegal                        1440
Egypt                          1439
Nigeria                        1439
Côte d'Ivoire                  1437
Uganda                         1423
Madagascar                     1415
Cameroon                       1415
Zambia                         1397
Ethiopia                       1393
South Africa                   1392
United Republic of Tanzania    1389
Mozambique                     1370
Malawi                         1368
Rwanda                         1360
Morocco                        1360
Angola                         1358
Botswana                       1349
Burkina Faso                   1346
Congo                          1342
Zimbabwe                       1340
Guinea                         1333
Mauritius                      1324
Niger                          1323
Ghana                          1313
Tunisia                        1295
Algeria                        1255
Namibia                     